mne/epochs.py

```diff
@@ -1252,14 +1258,14 @@ def next(self, return_event_id=False):
         """
         if self.preload:
             if self._current >= len(self._data):
-                raise StopIteration
```

Comment:
Apparently using `StopIteration` is a ["hack"] (https://www.python.org/dev/peps/pep-0479/) and we needed to refactor these functions, so I've done that here.